pkg/operator/ceph/controller/spec.go

```diff
@@ -394,9 +394,26 @@ func ContainerEnvVarReference(envVarName string) string {
 }
 
 // DaemonEnvVars returns the container environment variables used by all Ceph daemons.
-func DaemonEnvVars(image string) []v1.EnvVar {
+func DaemonEnvVars(cephClusterSpec *cephv1.ClusterSpec) []v1.EnvVar {
+
+	var rookMsgr2EnvValue string
+	if cephClusterSpec.Network.Connections != nil {
+		rookMsgr2EnvValue = fmt.Sprintf(
+			"msgr2_%t_encryption_%t_compression_%t",
+			cephClusterSpec.Network.Connections.RequireMsgr2,
+			cephClusterSpec.Network.Connections.Encryption.Enabled,
+			cephClusterSpec.Network.Connections.Compression.Enabled,
+		)
+	}
+
+	rookMsgr2Env := []v1.EnvVar{{
+		Name:  "ROOK_MSGR2",
+		Value: rookMsgr2EnvValue,
+	}}
+
+	cephDaemonsEnvVars := append(k8sutil.ClusterDaemonEnvVars(cephClusterSpec.CephVersion.Image), rookMsgr2Env...)
```

Comment:
How about only appending this env var if the network connections != nil? Then we won't cause all daemons to restart during the first upgrade with this setting even if they didn't enable msgr2.

Reply:
yes, we need to check `!=nil` for every condition, or else we'll get nil pointer error.